application/common/components/Emailer.php

```diff
@@ -382,11 +388,16 @@ public function sendMessageTo(
         $bccAddress = $data['bccAddress'] ?? '';
         $subject = $this->getSubjectForMessage($messageType, $dataForEmail);
 
+        if (empty($subject)) {
+            \Yii::error(sprintf(
+                'No subject known for %s email messages.',
+                $messageType
+            ));
+        }
```

Comment:
Can/should this check be done within `email`?

Reply:
Good question. I think it belongs here, since this is where the `getSubjectForMessage()` call is happening. When `email()` is called, it's being handed a pre-rendered subject. This warning is about when we're trying to look up the subject for a given message type, but no such subject was found. (I hit that situation before realizing I needed to add the new subjects to the `subjects` array internally; in that case, the subject just comes back as an empty string, which was not helpful.)

I suppose I could move it into `getSubjectForMessage()`... that actually may make more sense than where I have it here. I'll probably do that, now that you mention the question of where this check would best be placed.